### PR TITLE
hisat 2 update: add sam output flags

### PR DESCRIPTION
**Pull request by astrovsky01**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Adds advanced sam flag options

### CODE REVIEW

**Comment by astrovsky01** [comment]
Based on https://github.com/galaxyproject/tools-iuc/issues/2569

File: tools/hisat2/hisat2.xml [hunk 1]
```diff
@@ -534,6 +554,36 @@ hisat2
...
             </conditional>
 
+            <conditional name="sam_options">
+                <param name="sam_options_selector" type="select" label="SAM options">
+                    <option value="defaults">Use default values</option>
+                    <option value="advanced">Specify output options</option>
+                </param>
+                <when value="defaults" />
+                <when value="advanced">
+                    <param name="no_unal" argument="--no-unal" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Suppress SAM records for reads that failed to align." help="Default: false"/>
```
Comment (bernt-matthias) [inline tools/hisat2/hisat2.xml:564]: Can you use lower case false?

**Comment by bgruening** [comment]
```
Output output_alignments:  different than expected, difference (using diff):
( /tmp/tmpvcdwekf5hisat_output_5.bam v. /tmp/tmpo7kfq79fhisat_output_5.bam )
Binary data detected, not displaying diff
```

I haven't had time to look closely at it, but it seems to be a problem with the BAM file. BAM files are unpacked automatically by Galaxy and can diffe'd, so this should be fine. So I would assume that maybe the lines_diff needs to be increased?

**Comment by astrovsky01** [comment]
Even at 40 lines different (more than twice the length of the whole converted sam file...), it's failing. When I run a samtools view + diff on the test and the output files, they only have one line different

**Comment by bernt-matthias** [comment]
Can you convert the bam file created during planemo test and the bam file in test-data, convert them manually to sam and compare the sam files? Maybe this sheds some light on the differences..

**Comment by bernt-matthias** [comment]
I did a bit more research on this. When I call my system installed version of `samtools view ` I get an error 

```
[E::sam_hrecs_error] Malformed key:value pair at line 3: "@RG	ID:read_group	TAG:test1	TAG:test2"
samtools view: failed to add PG line to the header
```

but if I use the `samtools` from the conda env it works. So I guess that older versions of `samtools` (and htslib?) might have some problems that have been fixed in later versions. 

Galaxy uses pysam to convert the bam to sam for calculating the diff. In the output of `planemo test` one can find: 

```
2021-01-23 13:53:45,189 WARNI [galaxy.tool_util.verify] Converting local (test-data) BAM to SAM failed: 'samtools returned with error 1: stdout=, stderr=samtools view: failed to add PG line to the header\n'. Will compare BAM files
```

So my guess is that we need a newer version of pysam for Galaxy...?

**Comment by bernt-matthias** [comment]
Think I found it, actually the read group data was wrongly formatted since only certain tags are allowed, see https://samtools.github.io/hts-specs/SAMv1.pdf page 4.

I would suggest to implement a validator to check for correct tags, or to add a select.

**Comment by bgruening** [comment]
Looks good to me. Hisat2 2.2.1 is out? Do we want to give it a try?

**Comment by astrovsky01** [comment]
Let's do that as a different PR -- this was just supposed to be the papercut

**Comment by bernt-matthias** [comment]
> I would suggest to implement a validator to check for correct tags, or to add a select.

Do you think that this would fit in the paper cut. At least we should add a short note in the help.

**Comment by astrovsky01** [comment]
Let's add that next so these flags can be in asap -- it hurts nothing to do these as separate PRs

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**